arangod/Agency/Node.h

```diff
@@ -75,6 +75,60 @@ typedef std::chrono::steady_clock::time_point SteadyTimePoint;
 
 class Store;
 
+class SmallBuffer {
+  uint8_t* _start;
+  size_t _size;
+ public:
+  SmallBuffer() : _start(nullptr), _size(0) {}
+  SmallBuffer(size_t size) : _start(new uint8_t[size]), _size(size) {}
+  SmallBuffer(SmallBuffer const& other) {
+    if (other.empty()) {
+      _start = nullptr;
+      _size = 0;
+    } else {
+      _start = new uint8_t[other._size];
+      _size = other._size;
+      memcpy(_start, other._start, other._size);
+    }
+  }
+  SmallBuffer(SmallBuffer&& other) {
+    _start = other._start;
+    other._start = nullptr;
+    _size = other._size;
+    other._size = 0;
+  }
+  SmallBuffer& operator=(SmallBuffer const& other) {
+    if (!empty()) {
+      delete[] _start;
```

Comment:
we could even optimize this case when `other` is not empty and we already have a buffer allocated. In case our own buffer is at least as large as the other buffer, we can copy the other's buffer into ours and save a `delete[]` and a `new[]`. However, that will also require tracking the buffer _capacity_ in addition to the buffer size, so it would be a somewhat larger change.

Reply:
I wanted to avoid exactly this and since we do not have realloc in C++, I chose against it.

Reply:
Fine